tools/base/checker.py

```diff
@@ -0,0 +1,285 @@
+import argparse
```

Comment:
This is really just a high-level observation, but we're building a lot of scaffolding here for running checkers; have you done some analysis of what else is out there in the Python ecosystem that might do something similar? It seems this could be a fairly standard problem, but I'm not familiar enough with the non-Google Python ecosystem to know here.